packages/loot-core/src/server/budget/util.js

```diff
@@ -20,3 +20,23 @@ export function unflatten2(arr) {
   return res;
 }
 
+// Note that we don't restrict values to `Number.MIN_SAFE_INTEGER <= value <= Number.MAX_SAFE_INTEGER`
+// where `Number.MAX_SAFE_INTEGER == 2^53 - 1` but a smaller range over `-(2^43-1) <= value <= 2^43 - 1`.
+// This ensure that the number is accurate not just for the integer component but for 3 decimal places also.
+//
+// This gives us the guarantee that can use `safeNumber` on number whether they are unscaled user inputs
+// or they have been converted to integers (using `amountToInteger`).
```

Comment:
I think we _can_ use `MAX_SAFE_INTEGER` and such because at this point we should always be working with integers. We shouldn't ever be doing arithmetic with floating points. Because we are only interested in integers we can expand the range we're OK with here.

Reply:
Yeah, ideally we always upscale and then check which would mean we can use `MAX_SAFE_INTEGER`. This is intended so that we fail early if we accidentally check then upscale then any value which would pass this check is still safe. These are both GDP-scale numbers so the extra range doesn't gain us much.

That said, we should be `safeNumber`ing any arithmetic so performed on these values so we'd likely get warnings anyway (except we wouldn't be able to tie it back to a particular cell).